CHANGELOG.md

```diff
@@ -1,5 +1,50 @@
 # Changelog
 
+## 0.21.0
+
+**Release date:** 2022-03-21
+
+This prerelease further improves git operations' stability, upgrades source
+controller to `v0.22` and prepares the code base for more standardized 
+controller runtime operations.
+
+The source-controller dependency was updated to version `v0.22` which 
+introduces API `v1beta2`. This set the minimum version for the 
+source-controller running in the same cluster to be `v0.22.0` or greater.
+
+Git operations using `go-git` have been migrated away to `git2go`, which is now
+the only framework this controller uses for interacting with repositories.
+
+A new experimental transport has been added to improve reliability, adding
+timeout enforcement for Git network operations.
+Opt-in by setting the environment variable `EXPERIMENTAL_GIT_TRANSPORT` to
+`true` in the controller's Deployment. This will result in the low-level
+transport being handled by the controller, instead of `libgit2`. It may result
+in an increased number of timeout messages in the logs, however it will remove
+the ability of Git operations to make the controllers hang indefinitely.
```

Comment:
Maybe instead of ability call it a bug?

Reply:
Thank you, @stefanprodan, it does make it clearer. :+1: